scripts/skills/perks/perk_rf_kingfisher.nut

```diff
@@ -35,82 +36,92 @@ this.perk_rf_kingfisher <- ::inherit("scripts/skills/skill", {
+			}
+		}
+
+		local netEffect = _targetEntity.getSkills().getSkillByID("effects.net");
+		netEffect.m.KingfisherPerk <- ::MSU.asWeakTableRef(this);
+		local onRemoved = netEffect.onRemoved;
+		netEffect.onRemoved = function()
+		{
+			onRemoved();
+			if (!::MSU.isNull(this.m.KingfisherPerk))
+			{
+				this.m.KingfisherPerk.setSpent(false);
+			}
+		}
+		local onDeath = netEffect.onDeath;
+		netEffect.onDeath = function( _fatalityType )
 		{
-			this.setSpent(true);
-
-			local self = this;
-			_targetEntity.getSkills().add(::MSU.new("scripts/skills/effects/net_effect", function(o) {
-				o.m.KingfisherPerk <- ::MSU.asWeakTableRef(self);
-				o.onRemoved <- function()
-				{
-					if (!::MSU.isNull(this.m.KingfisherPerk))
-					{
-						this.m.KingfisherPerk.setSpent(false);
-					}
-				}
-				o.onDeath <- function( _fatalityType )
-				{
-					this.onRemoved();
-				}
-			}));
-
-			_targetEntity.getSkills().add(::MSU.new("scripts/skills/actives/break_free_skill", function(o) {
-				o.m.Icon = "skills/active_74.png";
-				o.m.IconDisabled = "skills/active_74_sw.png";
-				o.m.Overlay = "active_74";
-				o.setChanceBonus(999);
-				o.m.SoundOnUse = [
-					"sounds/combat/break_free_net_01.wav",
-					"sounds/combat/break_free_net_02.wav",
-					"sounds/combat/break_free_net_03.wav"
-				];
-			}));
-
-			local isReinforced = this.getContainer().getSkillByID("actives.throw_net").m.IsReinforced;
-			local effect = ::Tactical.spawnSpriteEffect(isReinforced ? "bust_net_02" : "bust_net", this.createColor("#ffffff"), _targetEntity.getTile(), 0, 10, 1.0, _targetEntity.getSprite("status_rooted").Scale, 100, 100, 0);
-			local flip = !_targetEntity.isAlliedWithPlayer();
-			effect.setHorizontalFlipping(flip);
-			::Time.scheduleEvent(::TimeUnit.Real, 200, this.onNetSpawn.bindenv(this), {
-				TargetEntity = _targetEntity,
-				IsReinforced = isReinforced
-			});
+			onDeath();
+			this.onRemoved();
 		}
+
+		this.m.NetEffect = ::MSU.asWeakTableRef(netEffect);
+
+		_targetEntity.getSkills().getSkillByID("actives.break_free").setChanceBonus(999);
+
+		this.setSpent(true);
 	}
 
-	// Make it impossible to swap the offhand item when IsSpent is true
-	function getItemActionCost( _items )
+	function onUpdate( _properties )
 	{
-		if (!this.m.IsSpent) return null;
+		if (!this.m.IsSpent && this.isEnabled())
+			_properties.Reach += 2;
+	}
 
-		foreach (item in _items)
+	function onMovementFinished( _tile )
+	{
+		// Lose the net if you end up at any tile more than 1 distance away from the target you have trapped
+		if (this.m.IsSpent && !::MSU.isNull(this.m.NetEffect) && this.m.NetEffect.getContainer().getActor().getTile().getDistanceTo(_tile) > 1)
 		{
-			if (item != null && item.getSlotType() == ::Const.ItemSlot.Offhand)
-				return 99;
+			this.m.NetEffect.m.KingfisherPerk = null;
+			this.setSpent(false);
+			this.getContainer().getActor().getItems().unequip(this.getContainer().getActor().getOffhandItem());
 		}
 	}
 
 	function setSpent( _isSpent )
```

Comment:
As this is a newly introduced function it should be sorted to the bottom of this script.
I would then also introduce those functions with `// New Functions`

Like so:
```

// New Functions
	function setSpent( _isSpent )
	{
		[...]
	}
```

Reply:
I am not convinced for the need for this, especially in a completely new skill script file. If it was a hooked file then I could see _some_ use for that even there I am not fully convinced but it's relevant. For brand new scripts this seems unnecessary. Vanilla doesn't do it either for special functions inside certain skills.

Reply:
And I think was bad practice by Vanilla. 
Those new utility functions mixed in between the overwritten functions makes it harder to understand what a skill is actually doing.

Reply:
As far as I could understand based on the discussion in Discord, we didn't establish any rule for this. But generally in the future I'll try to put utility/helper functions at the bottom of the files.